src/main/java/uk/gov/hmcts/reform/prl/handlers/CaseFlagsEventHandler.java

```diff
@@ -6,52 +6,36 @@
-import uk.gov.hmcts.reform.idam.client.models.UserDetails;
-import uk.gov.hmcts.reform.prl.clients.RoleAssignmentApi;
 import uk.gov.hmcts.reform.prl.clients.ccd.records.StartAllTabsUpdateDataContent;
 import uk.gov.hmcts.reform.prl.enums.CaseEvent;
-import uk.gov.hmcts.reform.prl.enums.amroles.InternalCaseworkerAmRolesEnum;
+import uk.gov.hmcts.reform.prl.enums.YesOrNo;
 import uk.gov.hmcts.reform.prl.events.CaseFlagsEvent;
-import uk.gov.hmcts.reform.prl.models.roleassignment.getroleassignment.RoleAssignmentResponse;
-import uk.gov.hmcts.reform.prl.models.roleassignment.getroleassignment.RoleAssignmentServiceResponse;
-import uk.gov.hmcts.reform.prl.services.UserService;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.CaseData;
+import uk.gov.hmcts.reform.prl.services.caseflags.CaseFlagsWaService;
 import uk.gov.hmcts.reform.prl.services.tab.alltabs.AllTabServiceImpl;
-
-import java.util.List;
+import uk.gov.hmcts.reform.prl.utils.CaseUtils;
 
 @Component
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class CaseFlagsEventHandler {
-    private final UserService userService;
-    private final RoleAssignmentApi roleAssignmentApi;
-    private final AuthTokenGenerator authTokenGenerator;
     private final ObjectMapper objectMapper;
     private final AllTabServiceImpl allTabService;
+    private final CaseFlagsWaService caseFlagsWaService;
 
     @Async
     @EventListener
     public void triggerDummyEventForCaseFlags(final CaseFlagsEvent event) {
-        String caseId = String.valueOf(event.callbackRequest().getCaseDetails().getId());
-        UserDetails userDetails = userService.getUserDetails(event.authorisation());
-        RoleAssignmentServiceResponse roleAssignmentServiceResponse = roleAssignmentApi.getRoleAssignments(
-            event.authorisation(),
-            authTokenGenerator.generate(),
-            null,
-            userDetails.getId()
-        );
 
-        List<String> roles = roleAssignmentServiceResponse
-            .getRoleAssignmentResponse()
-            .stream()
-            .map(RoleAssignmentResponse::getRoleName)
-            .toList();
-        if (roles.stream().anyMatch(InternalCaseworkerAmRolesEnum.CTSC.getRoles()::contains)) {
+        CaseData caseData = CaseUtils.getCaseData(event.callbackRequest().getCaseDetails(), objectMapper);
+        if (caseData.getReviewRaRequestWrapper() != null
+            && YesOrNo.No.equals(caseData.getReviewRaRequestWrapper().getIsCaseFlagsTaskCreated())
+            && !caseFlagsWaService.isCaseHasNoRequestedFlags(caseData)) {
+            String caseId = String.valueOf(caseData.getId());
             StartAllTabsUpdateDataContent startAllTabsUpdateDataContent = allTabService.getStartUpdateForSpecificEvent(
                 caseId,
                 CaseEvent.CREATE_WA_TASK_FOR_CTSC_CASE_FLAGS.getValue()
             );
-
+            startAllTabsUpdateDataContent.caseDataMap().put("isCaseFlagsTaskCreated", YesOrNo.Yes);
```

Comment:
should we make sure all these map keys (names) are setup as static fields somewhere. So  its easier to find them/their uses